### ISSUE TITLE
Parameter to ignore very first epochs in ModelCheckpoint

### ISSUE THREAD

**Issue by Kaveh8** [keras-team/keras#15260]
I think it will be good if you add an argument to some built-in callbacks. i.e. `ModelCheckpoint` to ignore the very first epochs. Mostly in first epochs developers don't want to apply any action, since the model may not be in a desirable level.

For example, while using `ModelCheckpoint` with `save_best_only` parameter, I do not want to apply this on the first 5 epochs. My model starts from zero,and it takes 5 epochs to go above a desirable threshold, then I want it to monitor the passed metric. 

It also could be a `threshold` parameter for the `monitor` metric to start applying the callback. 

Code Example:
```
model_checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='val_accuracy',
    mode='max',
    threshold = 0.50,            # for instance
    save_best_only=True)
```
Then it will start monitoring after the `val_accuracy` goes above the passed `threshold`. 

or
```
model_checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(
    filepath=checkpoint_filepath,
    save_weights_only=True,
    monitor='val_accuracy',
    mode='max',
    ignore_firsts = 5,            # for instance
    save_best_only=True)
```
Then it will start monitoring after the 5 epochs. 

Thanks in advance.

**Comment by rchao** [comment]
Thanks for the issue. Since ModelCheckpoint already has a wealth of complexity, will it work for you if you use a wrapper (custom) callback where it only calls ModelCheckpoint's epoch methods when epoch >= 5 for example?

**Comment by Kaveh8** [comment]
@rchao  Thanks for your response. You mean that I can write a custom callback (wrapper) and in its `on_epoch_end` method, after checking the epoch number >=5, I call `ModelCheckpoint`'s epoch methods? How can I do the last part (calling ModelCheckpoint's methods)? Could you please give me an example?

**Comment by jvishnuvardhan** [comment]
@Kaveh8 I implemented a custom callback that demonstrate what @rchao suggested above. Please take a look at the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/f3bb9899132c41e1916d23e0c1efe1ed/untitled.ipynb). Thanks!

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by rchao** [comment]
Hello @Kaveh8, on second thought a subclass ModelCheckpoint may be even easier (as @jvishnuvardhan pointed out in the colab). Let us know if this helps. Thanks!

**Comment by Kaveh8** [comment]
@rchao  @jvishnuvardhan That was great. Thank you for your help.
